resources-service/migrations/20230223150400-create-resource-string-attributes-table.js

```diff
@@ -0,0 +1,120 @@
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+module.exports = {
+  up: async queryInterface => {
+    await queryInterface.sequelize.query(`
+      CREATE TABLE IF NOT EXISTS resources."ResourceStringAttributes"
+      (
+        "resourceId" text COLLATE pg_catalog."default" NOT NULL,
+        "accountSid" text COLLATE pg_catalog."default" NOT NULL,
+        "key" text COLLATE pg_catalog."default" NOT NULL,
+        "value" text COLLATE pg_catalog."default" NOT NULL,
+        "language" text COLLATE pg_catalog."default" NOT NULL,
+        "info" JSONB,
+        "lastUpdated" timestamp with time zone,
+        "updateSequence"  bigint NOT NULL DEFAULT nextval('"Resources_updates_seq"'::regclass),
+        CONSTRAINT "ResourceStringAttributes_pkey" PRIMARY KEY ("resourceId", "accountSid", "key", "language", "value"),
+        CONSTRAINT "FK_ResourceStringAttributes_Resources" FOREIGN KEY ("resourceId", "accountSid")
+            REFERENCES resources."Resources" (id, "accountSid") MATCH SIMPLE
+            ON UPDATE CASCADE
+            ON DELETE CASCADE
+      )
+    `);
+    console.log('Table "ResourceStringAttributes" created');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE IF EXISTS resources."ResourceStringAttributes"
+          OWNER to resources;
+    `);
+    console.log('Table "ResourceStringAttributes" now owned by resources');
+
+    await queryInterface.sequelize.query(`
+      CREATE FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"()
+        RETURNS trigger
+        LANGUAGE 'plpgsql'
+        NOT LEAKPROOF
+      AS $BODY$
+      BEGIN
+        IF TG_WHEN <> 'BEFORE' THEN
+          RAISE EXCEPTION 'ResourcesLookupTables_updateSequence_trigger() may only run as an BEFORE trigger';
+        END IF;
+
+        IF (TG_LEVEL <> 'ROW' OR (TG_OP <> 'UPDATE')) THEN
+          RAISE EXCEPTION 'ResourcesLookupTables_updateSequence_trigger() added as trigger for unhandled case: %, %',TG_OP, TG_LEVEL;
+          RETURN NULL;
+        END IF;
+        
+        NEW."updateSequence" = nextval('"Resources_updates_seq"'::regclass);
+        RETURN NEW;
+      END
+      $BODY$;
+    `);
+    console.log('Function "ResourcesLookupTables_updateSequence_trigger" created.');
+
+    await queryInterface.sequelize.query(
+      `ALTER FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"() OWNER TO resources`,
+    );
+    console.log('Function "ResourcesLookupTables_updateSequence_trigger" ownership altered.');
+
+    await queryInterface.sequelize.query(`
+      CREATE TRIGGER "ResourceStringAttributes_update_trigger"
+      BEFORE UPDATE
+      ON resources."ResourceStringAttributes"
+      FOR EACH ROW
+      WHEN (pg_trigger_depth() = 0)
+      EXECUTE FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"();
+    `);
+    console.log('Trigger ResourceStringAttributes_update_trigger created');
```

Comment:
We'll need a trigger like this to be executed on the future "resource-related tables" updates as well, right?

Can we just update this same trigger, like incrementing the line `BEFORE UPDATE ON table1, table2, table3`? Or we need to create a separate trigger per table?

Reply:
Postgres doesn't support the same trigger being applied to multiple tables. I think Postgres considers a TRIGGER to be something that's 'owned' or scoped to a particular table, so can't be associated with more than one

However, we can share the 'trigger function', i.e. `ResourcesLookupTables_updateSequence_trigger` - this same function can be invoked by several different triggers attached to several different tables, so that's how I plan to approach it, the triggers have to be duplicated but the function is shared